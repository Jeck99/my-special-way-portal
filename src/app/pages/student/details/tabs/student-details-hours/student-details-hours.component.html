<div class="msw-student-details-hours">
  מערכת שעות ################ CONTENT #########
</div>
