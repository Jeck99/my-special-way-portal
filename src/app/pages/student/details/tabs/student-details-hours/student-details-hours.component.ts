import { Component, OnInit } from '@angular/core';
import { ActivatedRoute } from '@angular/router';

@Component({
  selector: 'app-student-details-hours',
  templateUrl: './student-details-hours.component.html',
  styleUrls: ['./student-details-hours.component.scss'],
})
export class StudentDetailsHoursComponent {

  sub: any;
  idOrNew: string;

  constructor(
    private route: ActivatedRoute,
  ) { }
}
