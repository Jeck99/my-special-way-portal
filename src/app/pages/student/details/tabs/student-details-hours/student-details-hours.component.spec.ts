import { TestBed } from '@angular/core/testing';
import { NO_ERRORS_SCHEMA } from '@angular/core';
import { Platform } from '@angular/cdk/platform';
import { Observable } from 'rxjs-compat';
import {ActivatedRoute, Router, RouterModule, Routes} from '@angular/router';
import {StudentDetailsHoursComponent} from './student-details-hours.component';
import {StudentDetailsComponent} from '../../student-details.component';

describe('Student Details Hours Component', () => {
  let routerModuleMock: Partial<RouterModule>;

  beforeEach(async () => {

    routerModuleMock = {
      forRoot: jest.fn(),
      forChild: jest.fn(),
      navigate: jest.fn(),
    };

    const routes: Routes = [
      { path: '', redirectTo: 'student', pathMatch: 'full' },
      {
        path: '',
        children: [
          { path: 'student/:idOrNew', component: StudentDetailsComponent,
            children: [
              { path: 'hours', component: StudentDetailsHoursComponent},
            ],
          },
        ],
      },
    ];

    TestBed.configureTestingModule({
      imports: [
        RouterModule.forRoot(routes),
      ],
      declarations: [
        StudentDetailsComponent,
        StudentDetailsHoursComponent,
      ],
      providers: [
        { provide: Router, useValue: routerModuleMock },
        Platform,
      ],
      schemas: [NO_ERRORS_SCHEMA],
    });
  });

  describe('with _new_ student path', () => {
    beforeEach(async () => {

      TestBed.configureTestingModule({
        providers: [
          {
            provide: ActivatedRoute, useValue: {
            parent: {
              params: Observable.of({idOrNew: '_new_'}),
            },
          },
          },
        ],
      });
    });

    it('should render the component as described in snapshot', () => {
      const fixture = TestBed.createComponent(StudentDetailsHoursComponent);
      expect(fixture).toMatchSnapshot();
    });
  });
});
