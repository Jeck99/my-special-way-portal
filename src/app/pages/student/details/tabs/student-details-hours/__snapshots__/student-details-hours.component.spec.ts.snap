// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Student Details Hours Component with _new_ student path should render the component as described in snapshot 1`] = `
<app-student-details-hours
  route={[Function Object]}
>
  <div
    class="msw-student-details-hours"
  >
    
  Not implemented yet
  
  </div>
</app-student-details-hours>
`;
